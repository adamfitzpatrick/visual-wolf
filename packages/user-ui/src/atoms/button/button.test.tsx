import { MouseEventHandler, MouseEvent } from 'react';
import { Button, ButtonProps, ButtonType } from '.';
import { fireEvent, render, screen } from '@testing-library/react';
import { WifiOutlined } from '@ant-design/icons';

describe('Button', () => {
  let clickSpy: jest.Mock<MouseEventHandler<HTMLButtonElement>>;
  let props: ButtonProps;

  beforeEach(() => {
    clickSpy = jest.fn();
    props = {
      type: 'default',
      onClick: clickSpy
    }
  });

  function doRender() {
    return render(<Button { ...props }>Button</Button>);
  }

  test('should show the included children for labelling', () => {
    doRender();
    expect(screen.getByText('Button')).toBeInTheDocument();
  });

  test('should set class properly for default button type', () => {
    expect(doRender().container).toMatchSnapshot();
  });

  test('should include a title if one was provided', () => {
    props.title = 'Title';
    doRender();
    expect(screen.getByTitle('Title')).toBeInTheDocument();
  });

  test('should set class properly for primary button type', () => {
    props.type = 'primary';
    expect(doRender().container).toMatchSnapshot();
  });

  test('should set class properly for text button type', () => {
    props.type = 'text';
    expect(doRender().container).toMatchSnapshot();
  });

  test('should set class properly for link button type', () => {
    props.type = 'link';
    expect(doRender().container).toMatchSnapshot();
  });

  test('should set class properly for large button size', () => {
    props.size = 'large';
    expect(doRender().container).toMatchSnapshot();
  });

  test('should set class properly for small button size', () => {
    props.size = 'small';
    expect(doRender().container).toMatchSnapshot();
  });

  test('should add icon-only class when an icon is provided with no children', () => {
    props.icon = <WifiOutlined />
    expect(render(<Button { ...props } />).container).toMatchSnapshot();
  });

  test('should show a disabled button when disabled is true', () => {
    props.disabled = true;
    expect(doRender().container).toMatchSnapshot();
  });

  test('should invoke click handler when clicked', () => {
    doRender();
    fireEvent.click(screen.getByText('Button'));
    expect(clickSpy).toHaveBeenCalled();
  });
});
