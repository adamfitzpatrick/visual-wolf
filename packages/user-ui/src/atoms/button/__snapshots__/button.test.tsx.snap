// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should add icon-only class when an icon is provided with no children 1`] = `
<div>
  <button
    class="button default icon-only"
  >
    <span>
      <span
        aria-label="wifi"
        class="anticon anticon-wifi"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="wifi"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M723 620.5C666.8 571.6 593.4 542 513 542s-153.8 29.6-210.1 78.6a8.1 8.1 0 00-.8 11.2l36 42.9c2.9 3.4 8 3.8 11.4.9C393.1 637.2 450.3 614 513 614s119.9 23.2 163.5 61.5c3.4 2.9 8.5 2.5 11.4-.9l36-42.9c2.8-3.3 2.4-8.3-.9-11.2zm117.4-140.1C751.7 406.5 637.6 362 513 362s-238.7 44.5-327.5 118.4a8.05 8.05 0 00-1 11.3l36 42.9c2.8 3.4 7.9 3.8 11.2 1C308 472.2 406.1 434 513 434s205 38.2 281.2 101.6c3.4 2.8 8.4 2.4 11.2-1l36-42.9c2.8-3.4 2.4-8.5-1-11.3zm116.7-139C835.7 241.8 680.3 182 511 182c-168.2 0-322.6 59-443.7 157.4a8 8 0 00-1.1 11.4l36 42.9c2.8 3.3 7.8 3.8 11.1 1.1C222 306.7 360.3 254 511 254c151.8 0 291 53.5 400 142.7 3.4 2.8 8.4 2.3 11.2-1.1l36-42.9c2.9-3.4 2.4-8.5-1.1-11.3zM448 778a64 64 0 10128 0 64 64 0 10-128 0z"
          />
        </svg>
      </span>
    </span>
    <span />
  </button>
</div>
`;

exports[`Button should set class properly for default button type 1`] = `
<div>
  <button
    class="button default"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should set class properly for large button size 1`] = `
<div>
  <button
    class="button default large"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should set class properly for link button type 1`] = `
<div>
  <button
    class="button link"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should set class properly for primary button type 1`] = `
<div>
  <button
    class="button primary"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should set class properly for small button size 1`] = `
<div>
  <button
    class="button default small"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should set class properly for text button type 1`] = `
<div>
  <button
    class="button text"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button should show a disabled button when disabled is true 1`] = `
<div>
  <button
    class="button default disabled"
    disabled=""
  >
    <span>
      Button
    </span>
  </button>
</div>
`;
